browser/brave_rewards/extension_rewards_notification_service_observer.cc

```diff
@@ -112,11 +112,8 @@ void ExtensionRewardsNotificationServiceObserver::OnGetAllNotifications(
                   NotificationsType>
       notifications_list;
   for (auto& item : rewards_notifications_list) {
-    notifications_list.push_back(
-        extensions::api::rewards_notifications::OnGetAllNotifications::
-            NotificationsType());
-    auto& notifications_type =
-        notifications_list[notifications_list.size() - 1];
+    notifications_list.emplace_back();
```

Comment:
Is there any reason here to first push a default-constructed object and just then initialize it?
It would be better if we changed the constructor and ended up with:
```
notifications_list.emplace_back(item.id_, item.type_, item.timestamp_,
                                item.args_);
```

Reply:
We can't do that unfortunately because `NotificationsType` is generated by the JSON Schema Compiler.

Reply:
Right. I guess it should be just
```
NotificationsType notifications_type;
notifications_type.id = item.id_;
notifications_type.type = item.type_;
notifications_type.timestamp = item.timestamp_;
notifications_type.args = item.args_;
notifications_list.emplace_back(std::move(notifications_type));
```
then.

Reply:
Fixed.